Automattic/wp-calypso
# Rewind Credentials: Refactor credentials form so it can be used outside of site-settings

Currently, the credentials form lives in `site-settings/jetpack-credentials/credentials-form` and is not easily reusable in other parts of Calypso. This PR does several things to make the component reusable:
- Moves the component to `components/rewind-credentials-form` to make it easier to import anywhere in Calypso.
- Changes the form to simply accept a site ID and credentials role, instead of all of the form field values. The form will now query state itself and decide what credentials to show, if any.
- Design improvements and UI simplification, including the addition of an "advanced" settings section.
- Make the private key field submittable when no password is present.
- Do not require the `path` field. Instead, default it to `/` so the server can determine the path.
- Default the server host to the site slug that corresponds to the provided site id.

**New form design**
![screen shot 2018-01-26 at 11 01 53 am](https://user-images.githubusercontent.com/5528445/35448490-7057aa3c-0288-11e8-8e10-f77a32a355e2.png)

**Testing Instructions**

- Load up the credentials interface in Site Settings > Security and add, delete, update, credentials for a self-hosted site. Make sure all requests are successful and there are no UI inconsistencies.
- For good measure, check that credentials can be autoconfigured and revoked for a Pressable site.
